src/graph/graph.c

```diff
@@ -554,24 +574,73 @@ int Graph_ConnectNodes(Graph *g, NodeID src, NodeID dest, int r, Edge *e) {
 	GrB_Matrix_setElement_BOOL(tadj, true, dest, src);
 	GrB_Index I = src;
 	GrB_Index J = dest;
-	id = SET_MSB(id);
+	edgeId = SET_MSB(edgeId);
 	info = GxB_Matrix_subassign_UINT64   // C(I,J)<Mask> = accum (C(I,J),x)
 		   (
 			   relationMat,         // input/output matrix for results
 			   GrB_NULL,            // optional mask for C(I,J), unused if NULL
 			   _graph_edge_accum,   // optional accum for Z=accum(C(I,J),x)
-			   id,                  // scalar to assign to C(I,J)
+			   edgeId,              // scalar to assign to C(I,J)
 			   &I,                  // row indices
 			   1,                   // number of row indices
 			   &J,                  // column indices
 			   1,                   // number of column indices
 			   GrB_NULL             // descriptor for C(I,J) and Mask
 		   );
 	assert(info == GrB_SUCCESS);
+}
+
+int Graph_ConnectNodes(Graph *g, NodeID src, NodeID dest, int r, Edge *e) {
+
+	Node srcNode;
+	Node destNode;
+
+	assert(Graph_GetNode(g, src, &srcNode));
+	assert(Graph_GetNode(g, dest, &destNode));
+	assert(g && r < Graph_RelationTypeCount(g));
 
+	EdgeID id;
+	Entity *en = DataBlock_AllocateItem(g->edges, &id);
+	en->id = id;
+	en->prop_count = 0;
+	en->properties = NULL;
+	e->entity = en;
+	e->relationID = r;
+	e->srcNodeID = src;
+	e->destNodeID = dest;
+	_Graph_ConnectNodesInMatrix(g, id, src, dest, r);
 	return 1;
 }
 
+// Try to add relation matrix if not present - used for graph deserilization.
+static inline void _Graph_TryAddRelationMatrix(Graph *g, int r) {
+	if(r != GRAPH_NO_RELATION) {
+		uint relation_count = Graph_RelationTypeCount(g);
+		if(r >= relation_count) {
+			for(uint i = relation_count; i <= r; i++) Graph_AddRelationType(g);
+		}
+	}
+}
+
+// Set a given edge in the graph - Used for deserialization of graph.
+// Returns 1 if connection is formed, 0 otherwise.
+void Graph_SetEdge(Graph *g, EdgeID edge_id, NodeID src, NodeID dest, int r, Edge *e) {
+	GrB_Info info;
+
+	Entity *en = DataBlock_AllocateItemOutOfOrder(g->edges, edge_id);
```

Comment:
I know I've raised this point a few times, but to my mind https://github.com/RedisGraph/RedisGraph/blob/master/src/util/datablock/oo_datablock.h#L14 is a solid indicator that these functions don't belong in this file.